src/lib/Toast/Toast.svelte

```diff
@@ -0,0 +1,122 @@
+<script lang="ts">
+  import { fly } from 'svelte/transition';
+  import { defaultToastProperties, type ToastProperties } from './properties';
+  import { createEventDispatcher, onDestroy, onMount } from 'svelte';
+
+  export let properties: ToastProperties = defaultToastProperties;
+  const dispatch = createEventDispatcher();
+  const colorMap: Record<string, string> = {
+    success: '#24aa5a',
+    error: '#f04438',
+    info: '#87ceeb',
+    warn: '#f3a42d'
+  };
+
+  let showToast = false;
+  let timeout: ReturnType<typeof setInterval>;
+  let backgroundColor =  properties.type !== undefined ? colorMap[properties.type] : '#87ceeb';
+
+  function hideToast() {
+    showToast = false;
+  }
+
+  function handleAnimationEnd() {
+    dispatch('handleAnimationEnd');
+  }
+
+  onMount(() => {
+    showToast = true;
+    timeout = setInterval(hideToast, properties.duration);
+  });
+
+  onDestroy(() => {
+    clearInterval(timeout);
+  });
+
+</script>
+
+{#if showToast}
+  <div
+    class="toast"
+    style="background-color: var(--toast-background-color, {backgroundColor})"
+    in:fly={{ x: 500, y: 0, duration: 400 }}
+    out:fly={{ x: 500, y: 0, duration: 800 }}
+    on:outroend={handleAnimationEnd}
+  >
+    {#if properties.leftIcon}
+      <div class="toast-icon-container">
+        <div class="toast-icon-wrapper">
+          <img class="toast-icon" src={properties.leftIcon} alt="toast-icon" />
+        </div>
+        <div class="toast-message">
+          {properties.message}
+        </div>
+      </div>
+    {:else}
+      <div class="toast-message">
+        {properties.message}
+      </div>
+    {/if}
+
+    {#if properties.showRightIcon}
+      <div>
+        <div class="close-button" on:click={hideToast} on:keypress>
+          <img
+            class="toast-icon"
+            src="https://sdk.breeze.in/gallery/icons/close.svg"
+            alt="close-icon"
+          />
+        </div>
+      </div>
+    {/if}
+  </div>
+{/if}
+
+<style>
+  .toast {
+    padding: var(--toast-padding, 10px);
+    font-size: var(--toast-font-size, 14px);
+    height: var(--toast-height, auto);
+    border-radius: var(--toast-border-radius, 0px);
+    width: var(--toast-width, auto);
+    align-items: var(--toast-align-items, center);
+    margin: var(--toast-margin, 0px 10px 10px 10px);
+    justify-content: space-between;
+    z-index: 1000;
+    display: flex;
+    position: absolute;
+    left: 0;
+    right: 0;
+  }
+
+  .toast-message {
+    color: var(--taost-text-color, #fff);
+  }
+
+  .toast-icon-container {
+    display: flex;
+    align-items: center;
+    gap: 10px;
+  }
+
+  .toast-icon-wrapper {
+    width: 20px;
+    height: 20px;
+  }
+
+  .toast-icon {
+    height: 100%;
+    border-radius: 50%;
+  }
+
+  .close-button {
+    width: var(--toast-width, 20px);
+    height: var(--toast-height, 20px);
+    cursor: var(--toast-cursor, pointer);
+    gap: 6px;
```

Comment:
add css variables to all style values here

Reply:
updated.